drivers/sensor/wsen_pads_2511020213301/wsen_pads_2511020213301_trigger.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_pads_2511020213301
+
+#include <zephyr/logging/log.h>
+
+#include "wsen_pads_2511020213301.h"
+
+LOG_MODULE_DECLARE(WSEN_PADS_2511020213301, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Enable/disable data-ready interrupt handling */
+static inline int pads_2511020213301_setup_interrupt(const struct device *dev, bool enable)
+{
+	const struct pads_2511020213301_config *cfg = dev->config;
+	unsigned int flags = enable ? GPIO_INT_EDGE_TO_ACTIVE : GPIO_INT_DISABLE;
+
+	return gpio_pin_interrupt_configure_dt(&cfg->interrupt_gpio, flags);
+}
+
+/*
+ * Is called when an interrupt has occurred.
+ */
+static inline void pads_2511020213301_handle_interrupt(const struct device *dev)
+{
+	struct pads_2511020213301_data *data = dev->data;
+
+	/* Disable interrupt handling until the interrupt has been processed */
+	pads_2511020213301_setup_interrupt(dev, false);
+
+#if defined(CONFIG_WSEN_PADS_2511020213301_TRIGGER_OWN_THREAD)
+	k_sem_give(&data->sem);
+#elif defined(CONFIG_WSEN_PADS_2511020213301_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&data->work);
+#endif
+}
+
+/* Calls data-ready trigger handler (if any) */
+static void pads_2511020213301_process_interrupt(const struct device *dev)
+{
+	struct pads_2511020213301_data *data = dev->data;
+
+#ifdef CONFIG_WSEN_PADS_2511020213301_PRESSURE_THRESHOLD
+	struct sensor_trigger pressure_low_trigger = {
+		.chan = SENSOR_CHAN_PRESS,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_PADS_2511020213301_THRESHOLD_LOWER};
+	struct sensor_trigger pressure_high_trigger = {
+		.chan = SENSOR_CHAN_PRESS,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_PADS_2511020213301_THRESHOLD_UPPER};
```

Comment:
Don't create new trigger structs on the stack; store a pointer to the user's sensor trigger and pass it back